types/mongodb/index.d.ts

```diff
@@ -618,23 +618,60 @@ export interface Collection<TSchema = Default> {
     /** http://mongodb.github.io/node-mongodb-native/3.0/api/Collection.html#updateMany */
-    updateMany(filter: Object, update: Object, callback: MongoCallback<UpdateWriteOpResult>): void;
-    updateMany(filter: Object, update: Object, options?: CommonOptions & { upsert?: boolean }): Promise<UpdateWriteOpResult>;
-    updateMany(filter: Object, update: Object, options: CommonOptions & { upsert?: boolean }, callback: MongoCallback<UpdateWriteOpResult>): void;
+    updateMany(filter: FilterQuery<TSchema>, update: Object, callback: MongoCallback<UpdateWriteOpResult>): void;
+    updateMany(filter: FilterQuery<TSchema>, update: Object, options?: CommonOptions & { upsert?: boolean }): Promise<UpdateWriteOpResult>;
+    updateMany(filter: FilterQuery<TSchema>, update: Object, options: CommonOptions & { upsert?: boolean }, callback: MongoCallback<UpdateWriteOpResult>): void;
     /** http://mongodb.github.io/node-mongodb-native/3.0/api/Collection.html#updateOne */
-    updateOne(filter: Object, update: Object, callback: MongoCallback<UpdateWriteOpResult>): void;
-    updateOne(filter: Object, update: Object, options?: ReplaceOneOptions): Promise<UpdateWriteOpResult>;
-    updateOne(filter: Object, update: Object, options: ReplaceOneOptions, callback: MongoCallback<UpdateWriteOpResult>): void;
+    updateOne(filter: FilterQuery<TSchema>, update: Object, callback: MongoCallback<UpdateWriteOpResult>): void;
+    updateOne(filter: FilterQuery<TSchema>, update: Object, options?: ReplaceOneOptions): Promise<UpdateWriteOpResult>;
+    updateOne(filter: FilterQuery<TSchema>, update: Object, options: ReplaceOneOptions, callback: MongoCallback<UpdateWriteOpResult>): void;
     /** http://mongodb.github.io/node-mongodb-native/3.0/api/Collection.html#watch */
     watch(pipeline?: Object[], options?: ChangeStreamOptions & { session?: ClientSession }): ChangeStream;
 }
 
+type FilterQuery<T> = {
+    [P in keyof T]?: T[P] | {
+        $eq?: T[P];
+        $gt?: T[P];
+        $gte?: T[P];
+        $in?: T[P][];
+        $lt?: T[P];
+        $lte?: T[P];
+        $ne?: T[P];
+        $nin?: T[P][];
+        $and?: (FilterQuery<T[P]> | T[P])[];
+        $not?: (FilterQuery<T[P]> | T[P])[] | T[P];
+        $expr?: any;
+        $jsonSchema?: any;
+        $mod?: [number, number];
+        $regex?: RegExp;
+        $options?: string;
+        $text?: {
+            $search: string;
+            $language?: string;
+            $caseSensitive?: boolean;
+            $diacraticSensitive?: boolean;
+        };
+        $where: Object;
+        $geoIntersects?: Object;
+        $geoWithin?: Object;
+        $near?: Object;
+        $nearSphere?: Object;
+        $elemMatch?: Object;
+        $size?: number;
+        $bitsAllClear?: Object;
+        $bitsAllSet?: Object;
+        $bitsAnyClear?: Object;
+        $bitsAnySet?: Object;
+    };
+};
```

Comment:
Also here, I think that  ` | { [key: string]: any };` is needed, since at the moment this type does not support filter of the type `{ $or: [{name: 'foo'}, {name: 'bar'}] }`